app/src/main/res/layout/layout_searchfragment.xml

```diff
@@ -1,17 +1,60 @@
 <?xml version="1.0" encoding="utf-8"?>
-<layout xmlns:tools="http://schemas.android.com/tools"
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
     xmlns:app="http://schemas.android.com/apk/res-auto">
-    <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+
+    <data>
+        <variable
+            name="searchViewModel"
+            type="com.guardian.podxdemo.presentation.search.SearchViewModel" />
```

Comment:
Ok, so passing the whole viewmodel in is a bit reckless. I also think the practise of wrapping the whole observable state in one data class is an excellent idea, although directly referencing it in the binding variable might have some performance overheads, as apparently each time the class is updated all the binding expressions in the view will be re-evaluated. Admittedly in this particular case this wouldn't actually be an issue since the only binding expression in that layout is being used is for the search string itself.

What i decided to do was define the string Variable in the data binding like so:

```xml
    <data>
        <variable
            name="search"
            type="String"/>
    </data>
```

And then create a (somewhat redundant) UiModel object that contains all the state that is to be exposed from the ViewModel, in this case just the search results:

```kotlin
data class SearchUiModel(
    val results: LiveData<List<SearchResult>>
)
```

This way even as the bound data and view model grows discrete pieces of data can be bound/observed separately to avoid unnecessary updates, although it would add a lot of code to the Fragment. Adding the search string to the data binding should make it easy to get the value of the edit text at any time from the fragment without having to use a `TextWatcher`, or directly reference the `EditText.text` field, not sure if it's entirely necessary though.